amip_interp/include/amip_interp.inc

```diff
@@ -16,382 +16,44 @@
- real :: teq  = 305. !< parameters for prescribed zonal sst option
- real :: tdif = 50. !< parameters for prescribed zonal sst option
- real :: tann = 20. !< parameters for prescribed zonal sst option
- real :: tlag = 0.875 !< parameters for prescribed zonal sst option
-
-
- integer :: amip_date(3)=(/-1,-1,-1/) !< amip date for repeating single day (rsd) option
-
- real :: sst_pert = 0. !< global temperature perturbation used for sensitivity experiments
-
- character(len=6) :: sst_pert_type = 'fixed'  !< use 'random' or 'fixed'
- logical :: do_sst_pert = .false.
- logical :: use_daily = .false. !< if '.true.', give 'data_set = 'daily''
-
- logical :: use_ncep_sst = .false. !< SJL: During nudging:   use_ncep_sst = .T.;  no_anom_sst = .T.
-                                   !!      during forecast:  use_ncep_sst = .T.;  no_anom_sst = .F.
- logical ::  no_anom_sst = .true.  !< SJL: During nudging:   use_ncep_sst = .T.;  no_anom_sst = .T.
-                                   !!      during forecast:  use_ncep_sst = .T.;  no_anom_sst = .F.
- logical :: use_ncep_ice = .false. !< For seasonal forecast: use_ncep_ice = .F.
- logical :: interp_oi_sst = .false. !< changed to false for regular runs
- logical :: use_mpp_io = .false. !< Set to .true. to use mpp_io, otherwise fms2io is used
-
- namelist /amip_interp_nml/ use_ncep_sst, no_anom_sst, use_ncep_ice,  tice_crit, &
-                            interp_oi_sst, data_set, date_out_of_range,          &
-                            use_zonal, teq, tdif, tann, tlag, amip_date,         &
-                            ! add by JHC
-                            sst_pert, sst_pert_type, do_sst_pert,                &
-                            use_daily,                                           &
-                            ! end add by JHC
-                            verbose, i_sst, j_sst, forecast_mode,                &
-                            use_mpp_io
-
-!-----------------------------------------------------------------------
-
-contains
 
 ! modified by JHC
 !> Retrieve sea surface temperature data and interpolated grid
-subroutine get_amip_sst (Time, Interp, sst, err_msg, lon_model, lat_model)
-
+subroutine GET_AMIP_SST_ (Time, Interp, sst, err_msg, lon_model, lat_model)
    type (time_type),         intent(in)    :: Time !< Time to interpolate
-   type (amip_interp_type),  intent(inout) :: Interp !< Holds data for interpolation
-   real,                     intent(out)   ::  sst(:,:) !< Sea surface temperature data
+   type (AMIP_INTERP_TYPE_),  intent(inout) :: Interp !< Holds data for interpolation
+   real(AMIP_INTERP_KIND_),                     intent(out)   ::  sst(:,:) !< Sea surface temperature data
    character(len=*), optional, intent(out) :: err_msg !< Holds error message string if present
 
-   real, dimension(mobs,nobs) :: sice
+   real(AMIP_INTERP_KIND_), dimension(mobs,nobs) :: sice
+   real(AMIP_INTERP_KIND_), allocatable:: temp1(:,:), temp2(:,:)
```

Comment:
is it safe to move temp1 and temp2 to here?

Reply:
I think so, but just to err on the side of caution, I've added the `save` attribute in 3e0db621c989b957ad4506fa4abcfae1e8869b2e.